bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/editor/targetComponent/details/ConicNameEditor.scala

```diff
@@ -24,6 +24,15 @@ final class ConicNameEditor(date: HorizonsIO[Date]) extends JPanel with Telescop
 
   def ct: ConicTarget = spt.getTarget.asInstanceOf[ConicTarget]
 
+  /** A program to resolve the current target BY NAME and replace it. */
+  val lookup: HorizonsIO[Unit] =
+    for {
+      d  <- date
+      t0 <- HorizonsIO.delay(ct)
+      p  <- Horizons.lookupConicTargetByName(t0.getName, t0.getTag.unsafeToHorizonsObjectType, d)
+      _  <- HorizonsIO.delay(spt.setTarget(p._1))
+    } yield ()
+
```

Comment:
I just moved this up from below to avoid a "suspicious forward reference" warning.